devtools/gradle/src/main/java/io/quarkus/devtools/project/buildfile/GradleProjectBuildFile.java

```diff
@@ -35,17 +33,23 @@ public GradleProjectBuildFile(Project project, ExtensionCatalog catalog) {
 
         final List<Dependency> boms = boms();
 
-        final Set<ResolvedArtifact> resolvedArtifacts = project.getConfigurations()
-                .getByName(JavaPlugin.RUNTIME_CLASSPATH_CONFIGURATION_NAME).getResolvedConfiguration()
-                .getResolvedArtifacts();
-        final List<ArtifactCoords> coords = new ArrayList<>(boms.size() + resolvedArtifacts.size());
+        final List<ArtifactCoords> coords = new ArrayList<>();
         boms.forEach(d -> {
             coords.add(new ArtifactCoords(d.getGroup(), d.getName(), null, "pom", d.getVersion()));
         });
-        resolvedArtifacts.forEach(a -> {
-            coords.add(new ArtifactCoords(a.getModuleVersion().getId().getGroup(), a.getName(),
-                    a.getClassifier(), a.getExtension(), a.getModuleVersion().getId().getVersion()));
-        });
+        project.getConfigurations()
+                .getByName(JavaPlugin.IMPLEMENTATION_CONFIGURATION_NAME).getIncoming().getDependencies().forEach(d -> {
+                    if (!(d instanceof ModuleDependency)) {
+                        return;
+                    }
+                    final ModuleDependency module = (ModuleDependency) d;
+                    coords.add(new ArtifactCoords(module.getGroup(), module.getName(), module.getVersion()));
+                    // why is the following code does not return any artifact?
+                    //module.getArtifacts().forEach(a -> {
```

Comment:
I think this is normal. The implementation configuration won't contains any resolved dependencies (where the `compilerClasspath` / `runtimeClasspath` configurations will). This configuration only goal is to be consumed not to contains resolved artifacts. 
However, the code above should works as expected by listing all dependency declared in the build file right?

Reply:
The problem is that we are missing the classifier and type of the artifacts. What the current impl assumes is the classifier is empty and the type is JAR.